hack/tools/go.mod

```diff
@@ -5,14 +5,11 @@ go 1.19
 require (
 	github.com/golangci/golangci-lint v1.50.0
 	github.com/joelanford/go-apidiff v0.1.0
-	github.com/onsi/ginkgo/v2 v2.8.3
-	k8s.io/code-generator v0.25.0
-	sigs.k8s.io/cluster-api/hack/tools v0.0.0-20211104153216-fb1f86267fed
+	github.com/onsi/ginkgo/v2 v2.9.7
+	k8s.io/code-generator v0.27.2
```

Comment:
Shouldn't matter that those got bumped in hack/tools as a consequence of bumping cluster-api/hack/tools.

I bumped cluster-api/hack/tools as the newer version of the release note tool is better